pandas/core/indexes/base.py

```diff
@@ -259,14 +259,7 @@ def _outer_indexer(self, left, right):
     # Constructors
 
     def __new__(
-        cls,
-        data=None,
-        dtype=None,
-        copy=False,
-        name=None,
-        fastpath=None,
-        tupleize_cols=True,
-        **kwargs,
+        cls, data=None, dtype=None, copy=False, name=None, tupleize_cols=True, **kwargs,
```

Comment:
Not sure how we historically handle but unfortunate this accepts kwargs - I suppose if someone keeps supplying `fastpath` nothing will change. Not sure if we should explicitly raise now in that case

Reply:
I changed `test_deprecated_fastpath` so instead of checking for a FutureWarning, it checks for a TypeError